### PR TITLE
switch order of list check to account for multiple SBO entries

### PR DESCRIPTION
**Pull request by gregmedlock**
Another small change in annotation checking--the previous line would lead to missing SBO terms for any object with more than one SBO term (e.g. it would end up trying things like ```['SBO:0000176', 'SBO:0000185'] in ['SBO:0000185']```).

### CODE REVIEW

**Comment by gregmedlock** [comment]
Related to this and my previous PR (#586), it looks like annotation/SBO could use some additional tests. I'm happy to help brainstorm a bit of what is needed and implement them as I continue to work through some of these edge cases.

**Comment by ChristianLieven** [comment]
Hi Greg! What were you thinking of? In general, I'm all ears for ideas on additional tests, but I'd also like  to point to some existing issues that need some love:

SBO:
- #469
- #522
- #537

Annotation:
- #315
- #187
- #186

File: memote/support/sbo.py [hunk 1]
```diff
@@ -67,4 +67,4 @@ def check_component_for_specific_sbo_term(items, term):
     return [elem for elem in items if
             elem.annotation is None or
             'sbo' not in elem.annotation or
-            elem.annotation['sbo'] not in term]
+            term not in elem.annotation['sbo']]
```
Comment (ChristianLieven) [inline memote/support/sbo.py]: Unfortunately, I don't think this fixes the issue entirely: 
- `test_transport_reaction_specific_sbo_presence` checks for a list of possible SBO terms related to transport reactions i.e. `term` can be a list of strings.
- The tests don't seem to cover that specific case

Would you be OK with taking that on? I realize that this is more than you bargained for.

Comment (ChristianLieven) [reply memote/support/sbo.py]: Really the whole SBO is a temporary stand-in for a solution handling the ontology as suggested in #469

**Comment by gregmedlock** [comment]
re: the issue with `test_transport_reaction_specific_sbo_presence`, it looks like that specific case is covered if any of the SBO transport terms are in `elem.annotation['sbo']`. I've updated the PR to check whether the terms are a list and handle accordingly. Seems a bit brute force but it gets the job done.

If more complex checks arise, e.g. check that multiple SBO terms are _all_ in an element's annotation, I think another function is warranted (e.g. `support.sbo.check_component_for_multiple_sbo_terms`).

I'll add tests for the changes in this PR and continue the higher level discussion in the other issues (#469 should probably come first).

In my second commit I accidentally slipped in an update to fix the gene annotation compliance check. If OK, I'll keep it in this PR.

**Comment by gregmedlock** [comment]
One more clarification... I was planning on adding tests for the gene annotation compliance checks, but wanted to make sure I wasn't missing them somewhere. My intuition is that they should be located in `test_for_annotation.py` but just haven't been implemented yet. If this is correct, please let me know and I'll add the gene-equivalent of the existing metabolite/reaction tests within that file first thing tomorrow. Thanks!

**Comment by ChristianLieven** [comment]
> One more clarification... I was planning on adding tests for the gene annotation compliance checks, but wanted to make sure I wasn't missing them somewhere. My intuition is that they should be located in test_for_annotation.py but just haven't been implemented yet.

1. Supporting functions for the `memote tests`, or `memote checks` if you prefer, go in `memote.support`,
2. The `memote tests` go in `memote.suite.tests`,
3. And `unit tests` for the support functions go in `tests.test_for_support`.

If I understand you correct you are talking about `unit tests` and in that case you are right to put them in `test_for_annotation`.

File: tests/test_for_support/test_for_sbo.py [hunk 2]
```diff
@@ -70,10 +81,18 @@ def test_find_components_without_sbo_terms(model, num, components):
     ("specific_sbo_term", 0, "metabolites"),
     ("specific_sbo_term", 1, "reactions"),
     ("specific_sbo_term", 1, "genes")
```
Comment (ChristianLieven) [inline tests/test_for_support/test_for_sbo.py]: A missing comma here leads to this error you're seeing on Travis:

```
==================================== ERRORS ====================================
___________ ERROR collecting tests/test_for_support/test_for_sbo.py ____________
tests/test_for_support/test_for_sbo.py:84: in <module>
    ("multiple_sbo_terms", 0, "metabolites"),
E   TypeError: 'tuple' object is not callable
```

File: tests/test_for_support/test_for_sbo.py [hunk 3]
```diff
@@ -70,10 +81,18 @@ def test_find_components_without_sbo_terms(model, num, components):
...
+    ("multiple_sbo_terms", 0, "metabolites"),
+    ("multiple_sbo_terms", 1, "reactions"),
+    ("multiple_sbo_terms", 1, "genes")
 ], indirect=["model"])
 def test_find_components_without_specific_sbo_term(model, num, components):
```
Comment (ChristianLieven) [inline tests/test_for_support/test_for_sbo.py]: I think it may be cleaner to take the `term` variable up into the parameters and only have a single assert statement in the test.

**Comment by gregmedlock** [comment]
I think we're good to go with the latest commit!
